### PR TITLE
[3.x] Enable method type information on release builds

### LINKED ISSUE #59723: [3.5 beta3] GDScript release builds can't infer type of builtin function return values
### Godot version

3.5 beta3

### System information

Linux, GLES 3, Intel Graphic UHD 630

### Issue description

Game crashes when below gdscript is compiled in release mode (in debug mode it can run normally)
```
extends Node

const MIN_TIME = 800
var start_time: int

func _ready():
	start_time = OS.get_ticks_msec()

func _process(_delta):
	var spent_time := OS.get_ticks_msec() - start_time
	var sleep_time = (MIN_TIME - spent_time) as float / 1000.0
```

with error message:
```
SCRIPT ERROR: Parse Error: The assigned value doesn't have a set type; the variable type can't be inferred.
          at: GDScript::load_byte_code (res://Main/test.gdc:11)
ERROR: Method failed. Returning: ERR_PARSE_ERROR
   at: load_byte_code (modules/gdscript/gdscript.cpp:782)
ERROR: Cannot load byte code from file 'res://Main/test.gdc'.
   at: load (modules/gdscript/gdscript.cpp:2205)
ERROR: Failed loading resource: res://Main/test.gdc. Make sure resources have been imported by opening the project in the editor at least once.
   at: _load (core/io/resource_loader.cpp:270)
ERROR: res://Main/Main.tscn:4 - Parse Error: [ext_resource] referenced nonexistent resource at: res://Main/test.gd
   at: poll (scene/resources/resource_format_text.cpp:412)
ERROR: Failed to load resource 'res://Main/Main.tscn'.
   at: load (core/io/resource_loader.cpp:206)
ERROR: Failed loading resource: res://Main/Main.tscn. Make sure resources have been imported by opening the project in the editor at least once.
   at: _load (core/io/resource_loader.cpp:270)
ERROR: Failed loading scene: res://Main/Main.tscn
   at: start (main/main.cpp:2083)
WARNING: ObjectDB instances leaked at exit (run with --verbose for details).
     at: cleanup (core/object.cpp:2064)
```

### Steps to reproduce

download the Minimal reproduction project, export game in release mode and run the game

### Minimal reproduction project

[test.zip](https://github.com/godotengine/godot/files/8387540/test.zip)

### PR DESCRIPTION
**Pull request by vnen**
This is the same as #53545 but for 3.x.

As a result, there is an increase in the binary size in release builds. For reference, these are the sizes I get in my machine (Intel macOS) compiled using `production=yes` and them stripped:
 
```
       3.x: 39 282 568 (37.46 MiB)
   This PR: 40 192 544 (38.33 MiB)
Difference:    909 976 ( 0.87 MiB)
```

Which is about a 2.32% increase in size. ~~The absolute increase is a bit bigger than the one in master which is likely because some of which I added here to release was already done in master before my PR.~~ Nevermind, it is way smaller now. I'll do the same on master to reduce the size there too.

This also removes the differences in type checking between debug and release for GDScript, which fixes #59723.

### CODE REVIEW

**Comment by Calinou** [comment]
Do you know if this type information could be made smaller somehow (e.g. by mangling argument names)?

File: core/method_bind.cpp [hunk 1]
```diff
@@ -34,20 +34,22 @@
...
-#ifdef DEBUG_METHODS_ENABLED
 PropertyInfo MethodBind::get_argument_info(int p_argument) const {
 	ERR_FAIL_INDEX_V(p_argument, get_argument_count(), PropertyInfo());
 
 	PropertyInfo info = _gen_argument_type_info(p_argument);
+#ifdef DEBUG_METHODS_ENABLED
 	info.name = p_argument < arg_names.size() ? String(arg_names[p_argument]) : String("arg" + itos(p_argument));
+#else
+	info.name = String("arg" + itos(p_argument));
+#endif
```
Comment (akien-mga) [inline core/method_bind.cpp:45]: I wonder if we could get away with not setting any name in the `#else` case. We care about the return type, but shouldn't need the `arg[0-9]` names. I guess that might have been what @neikeq referred to in the original PR.

Comment (vnen) [reply core/method_bind.cpp]: It's not what neikeq referred to, but I removed this since it's probably not needed.

**Comment by vnen** [comment]
I realized the `D_METHOD` was made a macro that ignores argument names when `DEBUG_METHODS_ENABLED` isn't defined, so my change added those to the binary. I now restored how it was before so the names are not included, which makes a lesser increase in size. I will update the OP with the new values.

**Comment by akien-mga** [comment]
Went from 9.5% increase to 2.3%, very nice!
That's definitely an acceptable tradeoff to fix issues related to type inference on release.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by lufog** [comment]
@akien-mga, after this pr, editor crash immediately on startup, with following backtrace:
```
Godot Engine v3.5.beta.custom_build.99c07c92e - https://godotengine.org
OpenGL ES 3.0 Renderer: NVIDIA GeForce GTX 1660/PCIe/SSE2
Async. shader compilation: OFF


================================================================
CrashHandlerException: Program crashed
Engine version: Godot Engine v3.5.beta.custom_build (99c07c92eb901e021de45a34158ba0004f5962c2)
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[0] embree::TaskScheduler::wait
[1] embree::TaskScheduler::wait
[2] main
[3] main
[4] widechar_main
[5] _main
[6] main
[7] embree::TaskScheduler::wait
[8] BaseThreadInitThunk
-- END OF BACKTRACE --
================================================================
```
Previous build (v3.5.beta.custom_build [385515431]) works fine.

**Comment by smix8** [comment]
Can confirm, (custom) builds compile without error but editor startup is completely broken on Windows with this / todays pr(s). The editor startup window crashes immediately which wasn't the case with yesterdays Windows build.

**Pull request merged**